packer/builder/azure/arm/storage_url.go

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See the LICENSE file in the project root for license information.
+
+package arm
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+type storageUrl struct {
+	url.URL
+	name      string
+	blob      string
+	container string
+	path      string
+	filename  string
+}
+
+func parseStorageUrl(rawurl string) (*storageUrl, error) {
```

Comment:
If I drop all of the other stuff then this class doesn't make much sense.  Consider it killed.